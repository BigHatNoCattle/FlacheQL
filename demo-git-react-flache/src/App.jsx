import React, { Component } from 'react'
import { render } from 'react-dom'
import GitBox from "./GitBox.jsx";
import QueryTimer from './QueryTimer.jsx';
import Flache from '../flache'
// import Flache from 'flacheql';

class App extends Component {
  constructor(props) {
    super(props);
    this.cache = new Flache();
    this.state = {
      gitBoxes: [],
      flacheTimer : {
        reqStartTime: null,
        lastQueryTime: 'Please wait...',
        timerText: 'Last query fetched 0 items in 0ms',
      },
      apolloTimer: {
        reqStartTime: null,
        lastQueryTime: 'Please wait...',
        timerText: 'Last query fetched 0 items in 0ms',
      },
      cache: this.cache
    };
    this.getBooksByAuthor = this.getBooksByAuthor.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.startTimer = this.startTimer.bind(this);
    this.endTimer = this.endTimer.bind(this);
    this.handleResponse = this.handleResponse.bind(this);
    this.buildBoxes = this.buildBoxes.bind(this);
  }

  componentDidMount() {
    this.getBooksByAuthor('react', '', 5, 10, true);
  }

  getBooksByAuthor(terms, language, stars, num, flache) {
    this.startTimer(flache, num);
    const endpoint = 'https://api.github.com/graphql'
    const headers = { "Content-Type": "application/graphql", "Authorization": "token d5db50499aa5e2c144546249bff744d6b99cf87d" }
    const query = this.buildQuery(terms, language, stars, num);
    // either fetch by flache or by apollo
    if (flache) {
      this.cache.it(query, endpoint, headers)
        .then(res => this.handleResponse(res.data, flache));
    } else {
      // use apollo cache/fetch method
      console.log('ALERT: Apollo functions not integrated!');
    }
  }

  buildQuery(terms, language, stars, num) {
    if (!num || num === 0) return window.alert('bad query! you must enter a number to search for!');
    if (!terms || terms === 'graphql');
    if (num > 100) return window.alert('max 100 results!');
    const searchQuery = `"${terms || ''}${language ? ' language:' + language : ''}${stars ? ' stars:>' + stars : ''}"`;
    if (searchQuery === '""') return window.alert('bad query! you must enter at least one filter!');
    return `{
      search(query: ${searchQuery}, type: REPOSITORY, first: ${num}) {
        repositoryCount
        edges {
          node {
            ... on Repository {
              name
              descriptionHTML
              stargazers {
                totalCount
              }
              forks {
                totalCount
              }
              updatedAt
            }
          }
        }
      }
    }`;
  }

  handleResponse(res, flache) {
    this.endTimer(flache, res.search.edges.length);
    this.buildBoxes(res);
  }

  buildBoxes(res) {
    const newBoxes = res.search.edges.map((repo, index) => {
      return <GitBox key={`b${index}`} name={repo.node.name} stars={repo.node.stargazers.totalCount} forks={repo.node.forks.totalCount}/>
    });
    this.setState({ gitBoxes: newBoxes });
  }

  startTimer(flache, num) {
    const reqStartTime = Date.now();
    const updatedTimer = { timerText: `Fetching ${num} items...`, reqStartTime, lastQueryTime: 'Please wait...' };
    // update either the flache or apollo timer
    if (flache) return this.setState({ flacheTimer: updatedTimer });
    return this.setState({ apolloTimer: updatedTimer });
  }

  endTimer(flache, num) {
    const lastQueryTime = flache ? `${Date.now() - this.state.flacheTimer.reqStartTime} ms` : `${Date.now() - this.state.apolloTimer.reqStartTime} ms`;
    // console.log('endTimer: lastQueryTime: ', lastQueryTime, '\n, flache?: ', flache);
    const updatedTimer = { timerText: `Last query fetched ${num} results in`, lastQueryTime, reqStartTime: null };
    // update either the flache or apollo timer
    if (flache) return this.setState({ flacheTimer: updatedTimer });
    return this.setState({ apolloTimer: updatedTimer });
  }

  handleSubmit(button) {
    const flache = button === 'flache';
    this.getBooksByAuthor(
      document.getElementById('searchText').value,
      document.getElementById('searchLang').value,
      Number(document.getElementById('searchStars').value),
      document.getElementById('searchNum').value,
      flache,
    );
  }

  render() {
    return (
      <div className="main-container">
        <div id="top-wrapper">
          <div id="form-wrapper">
            <h2>Find Github Repositories</h2>
            <div className="searchBoxes">
              <label>Search: <input id="searchText" type="text" className="text"/></label>
            </div>
            <div className="searchBoxes">
              <label>Language: <input id="searchLang" type="text" className="text"/></label>
            </div>
            <div className="searchBoxes">
              <label># of â˜†: <input id="searchStars" type="text" className="text"/></label>
            </div>
            <div className="searchBoxes">
              <label># to fetch: <input id="searchNum" type="text" className="text"/></label>
            </div>
            <input type="button" value="Search" onClick={() => this.handleSubmit('flache')} />
          </div>
          <QueryTimer
            lastQueryTime={this.state.flacheTimer.lastQueryTime}
            timerText={this.state.flacheTimer.timerText}
          />
        </div>
        <div className="result-list">
          {this.state.gitBoxes}
        </div>
      </div>
    )
  }
}

export default App;