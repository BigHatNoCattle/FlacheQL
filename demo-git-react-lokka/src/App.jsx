import React, { Component } from 'react'
import { render } from 'react-dom'
import GitBox from "./GitBox.jsx";
import QueryTimer from './QueryTimer.jsx';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      gitBoxes: [],
      reqStartTime: null,
      lastQueryTime: 'Please wait...',
      timerText: 'Last query fetched 0 items in 0ms',
    };
    this.getBooksByAuthor = this.getBooksByAuthor.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.startTimer = this.startTimer.bind(this);
    this.endTimer = this.endTimer.bind(this);
  }

  componentDidMount() {
    this.getBooksByAuthor('graphql', 'javascript', 5, 10);
  }

  getBooksByAuthor(terms, language, stars, num) {
    if (!num || num === 0) return window.alert('bad query! you must enter a number to search for!');
    if (!terms || terms === 'graphql');
    if (num > 100) return window.alert('max 100 results!');
    this.startTimer(num);
    const searchQuery = `"${terms || ''}${language ? ' language:' + language : ''}${stars ? ' stars:>' + stars : ''}"`;
    if (searchQuery === '""') return window.alert('bad query! you must enter at least one filter!');
    const query = `{
      search(query: ${searchQuery}, type: REPOSITORY, first: ${num}) {
        repositoryCount
        edges {
          node {
            ... on Repository {
              name
              descriptionHTML
              stargazers {
                totalCount
              }
              forks {
                totalCount
              }
              updatedAt
            }
          }
        }
      }
    }`;
    console.log('query to fetch: ', query);
    fetch('https://api.github.com/graphql', {
      method: 'POST',
      headers: { "Content-Type": "application/graphql", "Authorization": "token d5db50499aa5e2c144546249bff744d6b99cf87d" },
      body: JSON.stringify({
        query,
      }),
    })
      .then((res) => {
        return res.json();
      })
      .then((res) => {
        console.log(res.data.search.edges)
        this.endTimer(res.data.search.edges.length);
        const newBoxes = res.data.search.edges.map((repo, index) => {
          return <GitBox key={`b${index}`} name={repo.node.name} stars={repo.node.stargazers.totalCount} forks={repo.node.forks.totalCount}/>
        });
        this.setState({ gitBoxes: newBoxes });
      });
  }

  startTimer(num) {
    const reqStartTime = Date.now();
    this.setState({ timerText: `Fetching ${num} items...`, reqStartTime, lastQueryTime: 'Please wait...' });
  }

  endTimer(num) {
    const lastQueryTime = `${Date.now() - this.state.reqStartTime} ms`;
    this.setState({ timerText: `Last query fetched ${num} results in`, lastQueryTime });
  }

  handleSubmit() {
    this.getBooksByAuthor(
      document.getElementById('searchText').value,
      document.getElementById('searchLang').value,
      Number(document.getElementById('searchStars').value),
      document.getElementById('searchNum').value,
    );
  }

  render() {
    return (
      <div className="main-container">
        <div id="top-wrapper">
          <div id="form-wrapper">
            <h2>Search Repositories</h2>
            <div className="searchBoxes">
              <label>Search Terms: <input id="searchText" type="text" className="text"/></label>
            </div>
            <div className="searchBoxes">
              <label>Language: <input id="searchLang" type="text" className="text"/></label>
            </div>
            <div className="searchBoxes">
              <label># of âœ¡: <input id="searchStars" type="text" className="text"/></label>
            </div>
            <div className="searchBoxes">
              <label># to fetch: <input id="searchNum" type="text" className="text"/></label>
            </div>
            <input type="button" value="Search" onClick={this.handleSubmit} />
          </div>
          <QueryTimer
            lastQueryTime={this.state.lastQueryTime}
            timerText={this.state.timerText}
          />
        </div>
        <div className="result-list">
          {this.state.gitBoxes}
        </div>
      </div>
    )
  }
}

export default App;