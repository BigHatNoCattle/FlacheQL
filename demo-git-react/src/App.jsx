import React, { Component } from 'react'
import { render } from 'react-dom'
import GitBox from "./GitBox.jsx";

console.log(typeof GitBox);

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    console.log('fucking render! gitbox is a', typeof GitBox);
    return (
      <div className="main-container">
        {/* {gitBoxes} */}
        <GitBox/>
      </div>
    )
  }
}

export default App;