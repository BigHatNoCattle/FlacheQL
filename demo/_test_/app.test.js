// React;
import React from 'react';
// Files;
import App from './../src/App.jsx';
import Flache from '../flache';
// Testing
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

configure({ adapter: new Adapter() });


/**
 * Snapshot testing of initial render
 */
test('should render App component correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<App />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('input form taking in search terms, language, #, and stars renders correctly', () => {
  const tree = renderer
    .create(<div id="form-wrapper">
      <h2>Find Github Repositories</h2>
      <div className="searchBoxes">
        <label>Search: <input id="searchText" type="text" className="text"/></label>
      </div>
      <div className="searchBoxes">
        <label>Language: <input id="searchLang" type="text" className="text"/></label>
      </div>
      <div className="searchBoxes">
        <label># of â˜†: <input id="searchStars" type="text" className="text"/></label>
      </div>
      <div className="searchBoxes">
        <label># to fetch: <input id="searchNum" type="text" className="text"/></label>
      </div>
      <input type="button" value="Search" onClick={() => this.handleSubmit([''])} />
      <input type="button" value="Search w/createdAt" onClick={() => this.handleSubmit(['createdAt'])} />
      <input type="button" value="Search w/createdAt and databaseId" onClick={() => this.handleSubmit(['createdAt', 'databaseId'])} />
      <input type="button" value="Search w/databaseId" onClick={() => this.handleSubmit(['databaseId'])} />
      <input type="button" value="Show query cache" onClick={() => console.log(this.cache.comparisonCache, this.cache.cache)} />
    </div>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * Testing initial component statefulness
 */
it('Git box is instantiated as empty array upon load', () => {
  const wrapper = shallow(<App/>);
  expect(wrapper.instance().state.gitBoxes).toEqual([])
});

it('Cache is instantiated as Flache Cache', () => {
  const wrapper = shallow(<App/>);
  expect(wrapper.instance().cache).toEqual(new Flache());
});

////
it('Clicking search button with valid values makes AJAX call to the Github API', () => {
  const wrapper = shallow(<App/>);
  const instance = wrapper.instance();
  expect(instance.handleClick;
  instance.incrementCounter();
  expect(instance.state.count).toBe(1);
});