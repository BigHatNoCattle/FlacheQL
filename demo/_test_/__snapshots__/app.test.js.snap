// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input form taking in search terms, language, #, and stars renders correctly 1`] = `
<div
  id="form-wrapper"
>
  <h2>
    Find Github Repositories
  </h2>
  <div
    className="searchBoxes"
  >
    <label>
      Search: 
      <input
        className="text"
        id="searchText"
        type="text"
      />
    </label>
  </div>
  <div
    className="searchBoxes"
  >
    <label>
      Language: 
      <input
        className="text"
        id="searchLang"
        type="text"
      />
    </label>
  </div>
  <div
    className="searchBoxes"
  >
    <label>
      # of ☆: 
      <input
        className="text"
        id="searchStars"
        type="text"
      />
    </label>
  </div>
  <div
    className="searchBoxes"
  >
    <label>
      # to fetch: 
      <input
        className="text"
        id="searchNum"
        type="text"
      />
    </label>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Search"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Search w/createdAt"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Search w/createdAt and databaseId"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Search w/databaseId"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Show query cache"
  />
</div>
`;

exports[`should render App component correctly 1`] = `
<div
  className="main-container"
>
  <div
    id="top-wrapper"
  >
    <div
      id="form-wrapper"
    >
      <h2>
        Find Github Repositories
      </h2>
      <div
        className="searchBoxes"
      >
        <label>
          Search: 
          <input
            className="text"
            id="searchText"
            type="text"
          />
        </label>
      </div>
      <div
        className="searchBoxes"
      >
        <label>
          Language: 
          <input
            className="text"
            id="searchLang"
            type="text"
          />
        </label>
      </div>
      <div
        className="searchBoxes"
      >
        <label>
          # of ☆: 
          <input
            className="text"
            id="searchStars"
            type="text"
          />
        </label>
      </div>
      <div
        className="searchBoxes"
      >
        <label>
          # to fetch: 
          <input
            className="text"
            id="searchNum"
            type="text"
          />
        </label>
      </div>
      <input
        className="search-no-params"
        onClick={[Function]}
        type="button"
        value="Search"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Search w/createdAt"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Search w/createdAt and databaseId"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Search w/databaseId"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Show query cache"
      />
    </div>
    <div
      id="timer-wrapper"
    >
      <QueryTimer
        class="timerF"
        lastQueryTime="Please wait..."
        timerText="Last query fetched 0 results in"
        title="FlacheQL"
      />
      <QueryTimer
        class="timerF"
        lastQueryTime="Please submit query..."
        timerText="Last query fetched 0 results in"
        title="Apollo"
      />
    </div>
    <CacheNotifier
      showCacheHit={true}
    />
  </div>
  <div
    className="result-list"
  />
</div>
`;
